openapi/src/main/java/io/helidon/openapi/OpenAPISupport.java

```diff
@@ -116,6 +149,88 @@ public void configureEndpoint(Routing.Rules rules) {
                 .get(webContext, this::prepareResponse);
     }
 
+    static synchronized SnakeYAMLParserHelper<ExpandedTypeDescription> helper() {
+        if (helper == null) {
+            helper = SnakeYAMLParserHelper.create(ExpandedTypeDescription::create);
+            adjustTypeDescriptions(helper.types());
+        }
+        return helper;
+    }
+
+    static Map<Class<?>, ExpandedTypeDescription> buildImplsToTypes(SnakeYAMLParserHelper<ExpandedTypeDescription> helper) {
+        return helper.entrySet().stream()
+                .map(Map.Entry::getValue)
+                .collect(Collectors.toMap(ExpandedTypeDescription::impl, Function.identity()));
+    }
+
+    private static void adjustTypeDescriptions(Map<Class<?>, ExpandedTypeDescription> types) {
+        /*
+         * We need to adjust the {@code TypeDescription} objects set up by the generated {@code SnakeYAMLParserHelper} class
+         * because there are some OpenAPI-specific issues that the general-purpose helper generator cannot know about.
+         */
+
+        /*
+         * In the OpenAPI document, HTTP methods are expressed in lower-case. But the associated Java methods on the PathItem
+         * class use the HTTP method names in upper-case. So for each HTTP method, "add" a property to PathItem's type
+         * description using the lower-case name but upper-case Java methods and exclude the upper-case property that
+         * SnakeYAML's automatic analysis of the class already created.
+         */
+        ExpandedTypeDescription pathItemTD = types.get(PathItem.class);
+        for (PathItem.HttpMethod m : PathItem.HttpMethod.values()) {
+            pathItemTD.substituteProperty(m.name().toLowerCase(), Operation.class, getter(m), setter(m));
+            pathItemTD.addExcludes(m.name());
+        }
+
+        /*
+         * An OpenAPI document can contain a property named "enum" for Schema and ServerVariable, but the related Java methods
+         * use "enumeration".
+         */
+        Set.<Class<?>>of(Schema.class, ServerVariable.class).forEach(c -> {
+            ExpandedTypeDescription tdWithEnumeration = types.get(c);
+            tdWithEnumeration.substituteProperty("enum", List.class, "getEnumeration", "setEnumeration");
+            tdWithEnumeration.addPropertyParameters("enum", String.class);
+            tdWithEnumeration.addExcludes("enumeration");
+        });
+
+        /*
+         * SnakeYAML derives properties only from methods declared directly by each OpenAPI interface, not from methods defined
+         *  on other interfaces which the original one extends. Those we have to handle explicitly.
+         */
```

Comment:
Just a note to say I appreciate the comments!!

Reply:
Aim to please!